### ISSUE TITLE
Error pulling image (latest) from ericvh/arm64-ubuntu, ApplyLayer exit status 1 stdout:  stderr: operation not supported

### ISSUE THREAD

**Issue by derisavi** [docker/docker#16778]
I'm a newbie to docker. I'm running docker on an ARM64 board. I get the following error message. I've also attached output of "docker version" and "docker info". Please let me know how else I can help in investigating this. Thanks!

```
root@ubuntu:~# docker -D pull ericvh/arm64-ubuntu
Pulling repository ericvh/arm64-ubuntu
b1ceea5d197a: Error pulling image (latest) from ericvh/arm64-ubuntu, ApplyLayer exit status 1 stdout:  stderr: operation not supported 
8af13baf0b0d: Download complete 
8e162dfbd0e0: Download complete 
834d9d5b569f: Download complete 
a75b0c4b5d93: Download complete 
cac8f8146dea: Download complete 
89e088554a86: Download complete 
bf1ed9a61f29: Download complete 
5b612bb8dbd7: Download complete 
3bf1beb8995f: Download complete 
c5df03a0ea6e: Error downloading dependent layers 
FATA[0098] Error pulling image (latest) from ericvh/arm64-ubuntu, ApplyLayer exit status 1 stdout:  stderr: operation not supported 

root@ubuntu:~# docker version
Client version: 1.5.0
Client API version: 1.17
Go version (client): go1.4.2 gccgo (Ubuntu 5.1~rc1-0ubuntu1.1) 5.0.1 20150414 (prerelease) [gcc-5-branch revision 222102]
Git commit (client): a8a31ef
OS/Arch (client): linux/arm64
Server version: 1.5.0
Server API version: 1.17
Go version (server): go1.4.2 gccgo (Ubuntu 5.1~rc1-0ubuntu1.1) 5.0.1 20150414 (prerelease) [gcc-5-branch revision 222102]
Git commit (server): a8a31ef

root@ubuntu:~# docker info
Containers: 2
Images: 14
Storage Driver: devicemapper
 Pool Name: docker-252:5-528517-pool
 Pool Blocksize: 65.54 kB
 Backing Filesystem: extfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 785.4 MB
 Data Space Total: 107.4 GB
 Metadata Space Used: 1.442 MB
 Metadata Space Total: 2.147 GB
 Udev Sync Supported: true
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.90 (2014-09-01)
Execution Driver: native-0.2
Kernel Version: 4.1.0+
Operating System: Ubuntu 15.04
CPUs: 32
Total Memory: 31.37 GiB
Name: ubuntu
ID: HO3U:USMU:5HSS:PF3I:QNRU:A2S5:5ZCE:TRAE:IQ45:WS7D:XZ2F:SPZ2
```

**Comment by hqhq** [comment]
I've seem this before, and I think it's been fixed in 1.7.0, 1.5.0 is too old to be supported, with #16898 you can build the latest Docker on ARM64, can you update the Docker version and give it a try?

**Comment by shamiali2008** [comment]
Hi,

I am seeing a similar issue on 1.7.0-rc2(Error pulling image (latest) from ericvh/arm64-ubuntu, operation not supported). This is run on a real ARM64 board with linux kernel 4.16. Please find the version below:

root@linaro-developer:/opt/docker_v1/docker# ./bundles/1.7.0-rc2/dynbinary/docker-1.7.0-rc2 version
Client version: 1.7.0-rc2
Client API version: 1.19
Go version (client): go1.4.2 gccgo (Ubuntu 5.1~rc1-0ubuntu1) 5.0.1 20150414 (prerelease) [gcc-5-branch revision 222102]
Git commit (client): 7ddecf7
OS/Arch (client): linux/arm64
DEBU[0295] Calling GET /version                         
INFO[0295] GET /v1.19/version                           
Server version: 1.7.0-rc2
Server API version: 1.19
Go version (server): go1.4.2 gccgo (Ubuntu 5.1~rc1-0ubuntu1) 5.0.1 20150414 (prerelease) [gcc-5-branch revision 222102]
Git commit (server): 7ddecf7
OS/Arch (server): linux/arm64

This is how I start the docker daemon:
./bundles/1.7.0-rc2/dynbinary/docker-1.7.0-rc2 -d -D -s overlay &

And the pull request fails as below:

root@linaro-developer:/opt/docker_v1/docker# ./bundles/1.7.0-rc2/dynbinary/docker-1.7.0-rc2 -D pull ericvh/arm64-ubuntu
DEBU[0135] Calling POST /images/create                  
INFO[0135] POST /v1.19/images/create?fromImage=ericvh%2Farm64-ubuntu%3Alatest 
DEBU[0135] pulling image from host "docker.io" with remote name "ericvh/arm64-ubuntu" 
DEBU[0135] pinging registry endpoint https://index.docker.io/v1/ 
DEBU[0135] attempting v1 ping for registry endpoint https://index.docker.io/v1/ 
DEBU[0135] pulling v2 repository with local name "ericvh/arm64-ubuntu" 
DEBU[0135] pinging registry endpoint https://registry-1.docker.io/v2/ 
DEBU[0135] attempting v2 ping for registry endpoint https://registry-1.docker.io/v2/ 
DEBU[0135] hostDir: /etc/docker/certs.d/registry-1.docker.io 
DEBU[0136] Getting authorization for ericvh/arm64-ubuntu [pull] 
DEBU[0136] Pulling tag from V2 registry: "latest"       
DEBU[0136] [registry] Calling "GET" https://registry-1.docker.io/v2/ericvh/arm64-ubuntu/manifests/latest 
DEBU[0136] Getting bearer token with map[realm:https://auth.docker.io/token service:registry.docker.io] for  
DEBU[0136] hostDir: /etc/docker/certs.d/auth.docker.io  
DEBU[0136] hostDir: /etc/docker/certs.d/registry-1.docker.io 
DEBU[0137] image does not exist on v2 registry, falling back to v1 
DEBU[0137] pulling v1 repository with local name "ericvh/arm64-ubuntu" 
DEBU[0137] [registry] Calling GET https://index.docker.io/v1/repositories/ericvh/arm64-ubuntu/images 
DEBU[0137] hostDir: /etc/docker/certs.d/index.docker.io 
Pulling repository ericvh/arm64-ubuntu
DEBU[0137] Retrieving the tag list                      
DEBU[0137] hostDir: /etc/docker/certs.d/registry-1.docker.io 
DEBU[0138] Got status code 200 from https://registry-1.docker.io/v1/repositories/ericvh/arm64-ubuntu/tags 
DEBU[0138] Registering tags                             

b1ceea5d197a: Pulling image (latest) from ericvh/arm64-ubuntu, endpoint: https://registry-1.docker.io/v1/ 
DEBU[0138] Ancestry: [b1ceea5d197aea86d8a41dc406d6b9a8819bdfd6009110cd3ee0239da7d76307 c5df03a0ea6e66f3476f244a2daf010fb1c6da43af5ac93e58df569ff9d12235 3bf1beb8995f3eafc67b167d28067c1fff4a122d8fbe7d70d4df63865ff173bc 5b612bb8dbd790e7ca2caf82685b3b6366396d7b160dbdf66c9d28c48427fbbd bf1ed9a61f29bedb5c05ce757307df8ac5e2245150725836e687b1d04343711d 89e088554a86080184ef0333535a869d9475b45947b9b67adcc197f70e7fd554 cac8f8146dea5b74772c9610e5606f290a62eb58d46049712aabd7b507b9aca4 a75b0c4b5d9305f4bb7666297cf64730705c1124cb76bc94a56de488ddefaa2e 834d9d5b569ff99aa52760309493e944a4be72bc22a5037469a6acf31792d3de 8e162dfbb1ceea5d197a: Pulling dependent layers 
DEBU[0138] hostDir: /etc/docker/certs.d/registry-1.docker.io 
8e162dfbd0e0: Pulling metadata 
b1ceea5d197a: Error pulling image (latest) from ericvh/arm64-ubuntu, operation not supported 
DEBU[0139] hostDir: /etc/docker/certs.d/dseasb33srnrn.cloudfront.net 
8e162dfbd0e0: Error downloading dependent layers 
Error pulling image (latest) from ericvh/arm64-ubuntu, operation not supported

**Comment by floion** [comment]
hi @derisavi @hqhq @shamiali2008 is there any update on this issue? I am also getting this error on arm64 using kernel 4.1.15

**Comment by shamiali2008** [comment]
@floion If I remember correctly, the problem with me was that I was using a nfs mounted rootfs. When I switched that to a ext4 based rootfs, it disappeared.

**Comment by floion** [comment]
thanks for your reply @shamiali2008 ; in my case we are relying on btrfs but this is onyl visible on armv8 devices

**Comment by hqhq** [comment]
@floion This kind of error usually comes from kernel, I've seem that return from `system.Lgetxattr` several times, and it's because you lack of some kernel configs required by your filesystem. I'm not familiar with btrfs, you can debug your Docker and confirm this.

**Comment by agherzan** [comment]
Indeed. I just checked this and it fails here: https://github.com/docker/docker/blob/v1.10.3/pkg/archive/archive.go#L430 . I'll come back with additional information / fix.

**Comment by floion** [comment]
@hqhq apparently @agherzan traced it to the missing CONFIG_BTRFS_FS_POSIX_ACL in our kernel.
I think this should be checked by https://github.com/docker/docker/raw/master/contrib/check-config.sh as I used that tool when I initially got the error and it did not show any indication of the missing kernel config.

**Comment by hqhq** [comment]
@floion Yes I think we need to add that to `check-config.sh`, I added that part for ext3/4 when I first met this issue, but I don't use btrfs, so you can fire a PR for btrfs.

**Comment by agherzan** [comment]
There is also another issue here. The problem is that if this happens, the partition on which docker is running accumulates disk usage with every push. We need to look into this too cause this doesn't look sane.

**Comment by agherzan** [comment]
Ref https://github.com/moby/moby/commit/1c886a70df9820e58caa14789d9ad37c366026d9#diff-a7d4f99f6b6a9d7855588d5d2963f8a2
